### PR TITLE
chore: move flag generation to UI

### PR DESCRIPTION
**Pull request by ryanio**
#### What does it do?
Brings flag generation and related tests to UI.
Adds `chai` for test assertion
Closes https://github.com/ethereum/grid/issues/216

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   "ignore": {},
   "devDependencies": {
     "babel-register": "^6.26.0",
+    "chai": "^4.2.0",
```
Comment (marcgarreau) [inline package.json]: I'm reluctant to bring in another assertion lib when we've already got `jest` which provides `expect`. I'm guessing this was the path of least resistance and that grid shell used this. Do you think converting to `expect` is worth the effort?

Comment (marcgarreau) [reply package.json]: Actually, this is trivial. Knocking it out now.

Comment (ryanio) [reply package.json]: Oops didn't realize we had jest in there already. Happy to convert it if you haven't already.

Comment (marcgarreau) [reply package.json]: no worries! almost done.

**Pull request merged**